patchutils.go

```diff
@@ -23,12 +23,20 @@ func InterDiff(oldDiff, newDiff io.Reader) (string, error) {
 		return "", fmt.Errorf("parsing oldDiff: %w", err)
 	}
 
+	if len(oldFileDiffs) == 0 {
+		return "", fmt.Errorf("oldDiff: %w", ErrEmptyDiffFile)
+	}
+
 	newFileDiffs, err := diff.NewMultiFileDiffReader(newDiff).ReadAllFiles()
 
 	if err != nil {
 		return "", fmt.Errorf("parsing newDiff: %w", err)
 	}
 
+	if len(newFileDiffs) == 0 {
+		return "", fmt.Errorf("newDiff: %w", ErrEmptyDiffFile)
+	}
```

Comment:
```suggestion
	newFileDiffs, err := diff.NewMultiFileDiffReader(newDiff).ReadAllFiles()
	if err != nil {
		return "", fmt.Errorf("parsing newDiff: %w", err)
	}
	if len(newFileDiffs) == 0 {
		return "", fmt.Errorf("newDiff: %w", ErrEmptyDiffFile)
	}
```

same for the block above.